### PR TITLE
Add support for virtual workflow threads

### PR DESCRIPTION
**Pull request by Quinn-With-Two-Ns**
Add support for virtual workflow threads

Closes https://github.com/temporalio/sdk-java/issues/1448

### CODE REVIEW

**Review by cretz — COMMENTED** [review]
LGTM. Only main thing is if we can get the full existing test suite running against virtual thread-based workers/workflows to see if it could help us find interesting things.

File: temporal-sdk/build.gradle [hunk 1]
```diff
@@ -49,4 +79,40 @@ task testResourceIndependent(type: Test) {
         includeCategories 'io.temporal.worker.IndependentResourceBasedTests'
         maxParallelForks = 1
     }
+}
+
+// To test the virtual thread support we need to run a seprate test suite with Java 21
+testing {
+    suites {
+        virtualThreadTests(JvmTestSuite) {
```
[the remark this answers is not in the document]
Comment (Quinn-With-Two-Ns) [reply temporal-sdk/build.gradle:87]: > If we have tests on thread interruption or deadlock detection

Virtual threads don't behave any differently then platform threads here FWIW so I don't think it is providing a lot of value to double our CI time.

File: temporal-sdk/src/main/java/io/temporal/internal/worker/PollerOptions.java [hunk 2]
```diff
@@ -196,6 +205,7 @@ public PollerOptions build() {
   private final int pollThreadCount;
   private final Thread.UncaughtExceptionHandler uncaughtExceptionHandler;
   private final String pollThreadNamePrefix;
+  private final boolean useVirtualThreads;
```
[the remark this answers is not in the document]
Comment (Quinn-With-Two-Ns) [reply temporal-sdk/src/main/java/io/temporal/internal/worker/PollerOptions.java]: renamed to `usingVirtualThreads`

File: temporal-sdk/build.gradle (hunk 1 above)
[the remark this answers is not in the document]
Comment (cretz) [reply temporal-sdk/build.gradle:87]: We have a whole suite of different workflow logic/expectations, might as well confirm it works. I think we can run as a concurrent CI job to not double our time. I think the peace of mind is worth it (still can have virtual-thread specific tests like this).

[the remark this answers is not in the document]
Comment (Quinn-With-Two-Ns) [reply temporal-sdk/build.gradle:87]: Yes that what this test suite is for if there is specific tests we can add them to this test site, but the differences between platforms and virtual threads are not enough to justify running all tests twice

[the remark this answers is not in the document]
Comment (Quinn-With-Two-Ns) [reply temporal-sdk/build.gradle:87]: I can resist this though when we mark virtual threads as non experimental

[the remark this answers is not in the document]
Comment (Quinn-With-Two-Ns) [reply temporal-sdk/build.gradle:87]: >Maybe even if it wasn't on in CI, even the ability to run it (e.g. some flag that changes all test workers and whatever supported plumbing needed) and confirmation that we did run it once and it works would be great.

Yep I can do that

**Review by cretz — APPROVED** [review]
Approving, but would like at least one run of the test suite done with all virtual threads in all 5 ways enabled.

File: temporal-sdk/build.gradle (hunk 1 above)
[the remark this answers is not in the document]
Comment (Quinn-With-Two-Ns) [reply temporal-sdk/build.gradle]: >Maybe even if it wasn't on in CI, even the ability to run it (e.g. some flag that changes all test workers and whatever supported plumbing needed) and confirmation that we did run it once and it works would be great.

Done by adding another test suite called `testsWithVirtualThreads`

**Pull request merged**